woocommerce/woocommerce
# Do not show product card footer for Business Services products

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:


In the marketplace Discover tab, Business Services products can appear within groups like Discover our Favourites, that show product cards optimised for extensions. This causes Business Services products in these groups to show pricing information, which they shouldn't.

We're adding a new attribute to the featured product data on WooCommerce.com which will be used here to prevent the card footer from showing for Business Services products.



Closes https://github.com/Automattic/woocommerce.com/issues/21882



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. If https://github.com/Automattic/woocommerce.com/pull/21969 is not merged yet, check out that branch on your WooCommerce.com dev environment
2. Follow the steps in that PR to ensure there is a Business Services product in the Discover our Favorites group of featured products. Also make sure you have some Business Services products added to the Business Services group. Remember to flush your local WCCOM cache after making the changes.
3. Next you'll need to point your WC Core environment to your local WCCOM
  a. Add a mu-plugin to your WC Core wp-env (as suggested in p6JqRr-aWb-p2), and put these two filters in it:

```php
add_filter( 'http_request_host_is_external', '__return_true' );

add_filter( 'http_request_args', function ( $r ) {
	$r['reject_unsafe_urls'] = false;
	$r['sslverify']          = false;
	$r['timeout']            = 60;

	return $r;
} );
```

  b. Get the ID of wp-env's WordPress container.

<img width="500" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1647564/13509f82-962c-4f88-92ce-5f39222f2f64">

  c. Link the container to woocommerce.test in the Docker network.

```bash
docker network connect woocommerce.test <container ID>
```

4. Check out this branch
5. Apply [this patch](https://github.com/user-attachments/files/17414800/wccom-connect.patch) to your WC Core environment
6. Go to `/plugins/woocommerce` and run `pnpm --filter='@woocommerce/plugin-woocommerce' build`
7. Visit the [in-app marketplace](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions).
8. Confirm you see your Business Services product in the Discover our Favorites group, and that it doesn't show a price or "Free download" underneath the description text
![Extensions_‹_WooCommerce_‹_woocommerce](https://github.com/user-attachments/assets/3d9e5166-6388-4d41-815f-3274e1774189)

9. Confirm the cards in the "Services to help your business grow" group are still shown as Business Services cards
![Extensions_‹_WooCommerce_‹_woocommerce](https://github.com/user-attachments/assets/86b2eee0-6492-488d-b03f-a07cd6f681d9)

10. Test around this issue





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Prevent Business Services products from showing pricing information when they're displayed as regular product cards on the Extensions page.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
